src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -826,6 +829,33 @@ void ep_rt_aot_create_activity_id (uint8_t *activity_id, uint32_t activity_id_le
 #endif
 }
 
+ep_rt_thread_handle_t ep_rt_aot_thread_get_handle (void)
+{
+    ThreadStore::AttachCurrentThread();
```

Comment:
CoreCLR version of this method does not attach the thread. It will NULL when the thread was not attached to the runtime yet. Should this do the same?

Note that attaching the thread to the runtime is impactful/observable operation. It is disallowed on server GC threads - attaching server GC thread to the runtime is going to lead to a crash.

Reply:
Switched to GetCurrentThreadIfAvailable() to handle NULL. I followed the pattern in [eventtrace ](https://github.com/dotnet/runtime/blame/main/src/coreclr/nativeaot/Runtime/eventtrace_gcheap.cpp#L467)to avoid thread fields being uninitialized but didn't know on server GC cases and the expense.

Reply:
Filed #89649 to track the use of `AttachCurrentThread `in `ETW::GCLog::ForceGCForDiagnostics`